packages/backend/src/server/api/api-handler.ts

```diff
@@ -44,6 +49,30 @@ export default (endpoint: IEndpoint, ctx: Koa.Context) => new Promise<void>((res
 		}).catch((e: ApiError) => {
 			reply(e.httpStatusCode ? e.httpStatusCode : e.kind === 'client' ? 400 : 500, e);
 		});
+
+		// Log IP
+		if (user) {
+			const ip = ctx.ip;
+			const ips = userIpHistories.get(user.id);
+			if (ips == null || !ips.has(ip)) {
+				if (ips == null) {
+					userIpHistories.set(user.id, new Set([ip]));
+				} else {
+					ips.add(ip);
+				}
+				fetchMeta().then(meta => {
+					if (!meta.enableIpLogging) return;
```

Comment:
Since fetchMeta is a relatively expensive process, we do not want to run it whenever possible.

Reply:
I don't think this is true, since meta is cached and fetching is done in the background.
https://github.com/misskey-dev/misskey/blob/f997b7dff237bfd49e1e4dc8578a9b66bead6028/packages/backend/src/misc/fetch-meta.ts#L6-L7